create_mode.go

```diff
@@ -0,0 +1,52 @@
+package zk
+
+import "fmt"
+
+const (
+	FlagPersistent                  = 0
+	FlagEphemeral                   = 1
+	FlagSequence                    = 2
+	FlagEphemeralSequential         = 3
+	FlagContainer                   = 4
+	FlagTTL                         = 5
+	FlagPersistentSequentialWithTTL = 6
+)
+
+type createMode struct {
+	flag        int32
+	ephemeral   bool
+	sequential  bool
+	isContainer bool
+	isTTL       bool
+}
+
+func (cm *createMode) toFlag() int32 {
+	return cm.flag
+}
+
+// parsing a flag integer into the CreateMode needed to call the correct
+// Create RPC to Zookeeper.
+//
+// NOTE: This parse method is designed to be able to copy and paste the same
+// CreateMode ENUM constructors from Java:
+// https://github.com/apache/zookeeper/blob/master/zookeeper-server/src/main/java/org/apache/zookeeper/CreateMode.java
+func parseCreateMode(flag int32) (createMode, error) {
+	switch flag {
+	case FlagPersistent:
+		return createMode{0, false, false, false, false}, nil
```

Comment:
since it's all boolean values wdyt about being verbose when creating the struct here so reordering the struct doesn't break the logic

also since the flag index matches we can just pass it

```
createMode{
        flag: FlagEphemeral, // or flag
	isEphemeral: false,
	isSequential: false,
	isContainer: false,
	isTTL: false,
}
```

Reply:
for this one, I chose to match exactly what the Java class definition text is, it allows others to and reviewers to copy and paste the Java Enum verbatim.

i can see doing it more the Go way, but for this particular one i fine myself liking the property of the copy from the Java native client/server code.  

https://github.com/apache/zookeeper/blob/d12aba599233b0fcba0b9b945ed3d2f45d4016f0/zookeeper-server/src/main/java/org/apache/zookeeper/CreateMode.java#L29